lib/utils.js

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const utils = module.exports
+
+/**
+ * Adds the relevant CLM attrs(code.function and code.filepath) to span if
+ * code_level_metrics.enabled is true and if span exists
+ *
+ * Note: This is not like the other in agent CLM support.  Next.js is very rigid
+ * with its file structure and function names. We're providing relative paths to Next.js files
+ * based on the Next.js page.  The function is also hardcoded to align with the conventions of Next.js.
+ *
+ * @param {Object} config agent config
+ * @param {TraceSegment} segment active segment to add CLM attrs to
+ * @param {Object} attrs list of CLM attrs to add to segment
+ */
+utils.assignCLMAttrs = function assignCLMAttrs(config, segment, attrs) {
```

Comment:
Do we want to also add a unit tests here so we can assert the "don't add CLM stuff if there's no segment" part of the conditional, since that's not currently covered by the versioned tests?

Reply:
I can write unit tests